rclcpp/include/rclcpp/experimental/subscription_intra_process.hpp

```diff
@@ -102,6 +102,12 @@ class SubscriptionIntraProcess : public SubscriptionIntraProcessBase
 #endif
   }
 
+  ~SubscriptionIntraProcess()
+  {
+    rcl_ret_t ret = rcl_guard_condition_fini(&gc_);
+    (void)ret;
```

Comment:
Maybe we could use this kind of code.

```suggestion
    if (rcl_guard_condition_fini(&gc_) != RCL_RET_OK) {
      RCUTILS_LOG_ERROR_NAMED(
        "rclcpp",
        "failed to destroy guard condition: %s", rcl_get_error_string().str);
    }
```